utils/parse.js

```diff
@@ -31,7 +31,14 @@ exports.default = function parse(path, content, context) {
   // provide the `filePath` like eslint itself does, in `parserOptions`
   // https://github.com/eslint/eslint/blob/3ec436ee/lib/linter.js#L637
   parserOptions.filePath = path
-
+  
+  // @typescript-eslint/parser will parse the entire project with typechecking if you provide
+  // "project" or "projects" in parserOptions. Removing these options means the parser will
+  // only parse one file in isolate mode, which is much, much faster.
+  // https://github.com/benmosher/eslint-plugin-import/issues/1408#issuecomment-509298962
+  delete parserOptions.project;
```

Comment:
Seems like a dirty hack :)

Reply:
`@typescript-eslint/parser` does exactly what it is told to do.
If you give it `project` or `projects`, it will do an entire workspace parse so it can provide complete type information.

This is a bit of a hack, sure, but then again ESLint doesn't provide a mechanism for parsing files up-front / accessing parsed files, so `eslint-plugin-import` having to do a manual parse itself is a hack as well.

Reply:
Should we rise the feature proposal?)

Reply:
It's been talked about a bit by the eslint folks, but it's not high priority. We talked about it with them when we were adding type information to the parser.

Ultimately it's a pretty big edge case - the _vast_ majority of plugins (and parsers) do not need to break module boundaries.